packages/engine/src/proto.rs

```diff
@@ -225,6 +225,21 @@ pub struct ExperimentRunBase {
     pub project_base: ProjectBase,
 }
 
+impl ExperimentRunBase {
+    /// Returns `true` if the experiment uses Python init or has any Python behavior.
+    pub fn has_python_init_or_behavior(&self) -> bool {
```

Comment:
We may need the Python runner for other places in the future, could you change the name to something like `requires_python_runner`?